contracts/TokenFactory.sol

```diff
@@ -258,8 +253,7 @@ contract TokenFactory is ReentrancyGuard, Ownable {
         //slither-disable-next-line arbitrary-send-eth
 
         // store collateralByDay for winner detecting
-        uint256 currentDay = startOfCurrentDay();
-        collateralByDay[currentDay][tokenAddress] -= receivedETH;
+        collateralById[currentCompetitionId][tokenAddress] -= receivedETH;
```

Comment:
Note: same as above, however here  `collateralById[currentCompetitionId][tokenAddress]` might underflow, since it would start at 0 in competition id rounds which the token was not created (and sold) at